cudapoa/src/cudapoa_batch.hpp

```diff
@@ -156,8 +156,9 @@ class CudapoaBatch : public Batch
                                          num_nucleotides_copied_ * sizeof(uint8_t), cudaMemcpyHostToDevice, stream_));
         CGA_CU_CHECK_ERR(cudaMemcpyAsync(input_details_d_->window_details, input_details_h_->window_details,
                                          poa_count_ * sizeof(claragenomics::cudapoa::WindowDetails), cudaMemcpyHostToDevice, stream_));
+        /// ToDo may need to revise the following sizeof()
         CGA_CU_CHECK_ERR(cudaMemcpyAsync(input_details_d_->sequence_lengths, input_details_h_->sequence_lengths,
-                                         global_sequence_idx_ * sizeof(uint16_t), cudaMemcpyHostToDevice, stream_));
+                                         global_sequence_idx_ * sizeof(*input_details_h_->sequence_lengths), cudaMemcpyHostToDevice, stream_));
```

Comment:
I think this is the right way to do it! Using this for the rest of the `sizeof` calls in allocate_block.hpp would resolve the issue of types being out of sync right?

Reply:
yes, the only concern is for null pointers. but in `allocate_block.hpp` we know the pointer is not null.